openshift/ocm-agent
# OSD-10224 initial structure

### What type of PR is this?
Feature

### What this PR does / why we need it?
This PR adds some initial implementation structure for OSD-10224 which helps unblock a lot of devwork and overcomes a lot of the initial stumbling hurdles of 'how do we introduce/use dependency XYZ?'

- Introduces OCM client as a dependency and injects it into the webhook handler
- Webhook handler has basic abilities to scan for `ManagedNotifications` and send servicelogs based on them, and update the `ManagedNotification` template status

### Which Jira/Github issue(s) this PR fixes?

OSD-10224

### Special notes for your reviewer:
How to test:
- Build the image, push to your personal repo (`make build-image`)
- Replace/deploy an `ocmagent` CR on the cluster that runs your image instead of the `app-sre` image
- You *must* ensure that your cluster is running the latest version of `ocm-agent-operator` that has necessary RBAC fixes in it.
- Follow the quick test setup guide [0] to install a PrometheusRule and ManagedNotification CR on the cluster.
- Watch the `ocm-agent` logs, within 10-15 minutes it should receive an alert from alertmanager.
- Check the `managednotification` CR status, it should report that it sent a servicelog
- Within an hour, you should receive a servicelog email.

### Pre-checks (if applicable):
- [ ] Ran `make generate` command locally to validate code changes

[0] https://docs.google.com/document/d/1tejx2fbitbp6_fxRwMH6o1i4D3x4ooOv_wwStgkX3rQ/edit
